langfuse-core/src/index.ts

```diff
@@ -363,6 +363,8 @@ abstract class LangfuseCoreStateless {
     const items = queue.splice(0, this.flushAt);
     this.setPersistedProperty<LangfuseQueueItem[]>(LangfusePersistedProperty.Queue, queue);
 
+    this.processQueueItems(items, this.flushAt, this.flushInterval);
```

Comment:
In `queue.splice(0, this.flushAt);`, we are taking the items to send to our server from the queue and put them into `processQueueItems`. If these are 10 events, and after 6 we reached `BATCH_SIZE_LIMIT`, we are not putting the remaining events back into the queue. Hence, these events would be lost. 

`queue` is an array of `LangfuseQueueItem`. I think we can return the elements to be sent to the server and all the remaining events, which did not go beyond `MAX_MSG_SIZE`. The remaining events can be written back into the queue. When doing that, it would be important to keep the order of the events in place. If we take from the front of the queue, we should add back to the front as well in this case.